business_objects/project.py

```diff
@@ -60,6 +66,54 @@ def get_label_distribution(
         return values[0]
 
 
+def get_confidence_distribution(
+    project_id: str,
+    labeling_task_id: str,
+    data_slice_id: Optional[str] = None,
+    num_samples: Optional[int] = None,
+) -> List[float]:
+    query_filter = (
+        session.query(RecordLabelAssociation.confidence)
+        .join(
+            LabelingTaskLabel,
+            (RecordLabelAssociation.labeling_task_label_id == LabelingTaskLabel.id)
+            & (LabelingTaskLabel.project_id == RecordLabelAssociation.project_id),
+        )
+        .join(
+            LabelingTask,
+            (LabelingTask.id == LabelingTaskLabel.labeling_task_id)
+            & (LabelingTask.project_id == LabelingTaskLabel.project_id),
+        )
+        .filter(
+            LabelingTask.id == labeling_task_id,
+            RecordLabelAssociation.source_type
+            == enums.LabelSource.WEAK_SUPERVISION.value,
+        )
```

Comment:
```suggestion
        .filter(
            RecordLabelAssociation.project_id == project_id,
            LabelingTask.id == labeling_task_id,
            RecordLabelAssociation.source_type
            == enums.LabelSource.WEAK_SUPERVISION.value,
        )
```
Filter by project_id, I guess this should do it